dialogue-annotations/src/main/java/com/palantir/dialogue/annotations/Constants.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.annotations;
+
+import com.palantir.conjure.java.dialogue.serde.DefaultConjureRuntime;
+import com.palantir.conjure.java.dialogue.serde.Encodings;
+import com.palantir.dialogue.BodySerDe;
+
+public final class Constants {
+
+    public static final BodySerDe DEFAULT_BODY_SERDE =
+            DefaultConjureRuntime.builder().encodings(Encodings.json()).build().bodySerDe();
+
+    private Constants() {}
+}
```

Comment:
This shouldn't be public, I'd have probably constructed another `DefaultConjureRuntime.builder().encodings(Encodings.json()).build().bodySerDe()` within `InputStreamDeserializer` since it's not terribly expensive to create, and doing so is a one-time cost at class initialization time. Sharing one instance via package private access also works, but the naming is tough -- this isn't really a constant, more of a singleton.

Reply:
Out of curiosity, why do we want to keep such internals package-private? (I realize the rest does it and I should have thought of it, but I'm not sure I completely understand the reason for it)

I initially moved it back to private fields, but remembered why I initially went with a shared class, which is that this ensures we use the same parameters/setup for both of these. This way, if we want to push some change for one of them, we don't need to remember to find and update the others too, bringing us better consistency in the implementation.

Reply:
> Out of curiosity, why do we want to keep such internals package-private?

The less we expose to consumers, the greater freedom we retain to change and improve the implementation without a long, expensive migration campaign. If a custom deserializer implementation wanted to reference this class, we'd prefer they used the standard API that we're using here (`DefaultConjureRuntime.builder().//etc`) because we're already locked into supporting that one, so it doesn't expand the protected api surface.

Also, it's not entirely clear that a change to one of them should necessarily impact the other, there are lots of types of changes which would impact jackson ser/de that wouldn't have any impact on binary streaming, however it is possible we could get clever and break streaming if we aren't careful.

Reply:
Both of these make sense! Would you rather I move back to private fields in this case? I'm happy to do so as I just need to revert the last commit :)

Reply:
I think either way is fine :-] Happy to merge as is

Reply:
Sounds great, thank you for the iterations and sharing your knowledge!